File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/csharp/controls/boundaries.cs
using System;
class Router {
    public event Action Primary = delegate { };
    public event Action Secondary = delegate { };
    public void Install(Action f, Action g) {
        Primary += f; // @S1
        Secondary += g; // @S2
    }
    public void Remove(Action f) { Primary -= f; }
    public void Fire() { Primary(); } // @I1
}

File: apps/codemap-search/experiments/event-navigation-poc/language-corpus/regressions/examples/java/controls/boundaries.java
interface Work { void execute(); }
class Router {
    Runnable primary, secondary;
    void install(Runnable f, Runnable g) {
        primary = f; // @S1
        secondary = g; // @S2
    }
    void fire() {
        Runnable relay = primary::run;
        relay.run(); // @I1
        relay.toString(); // @I2
    }
    void custom() {
        Work relay = secondary::run;
        relay.execute(); // @I3
    }
}
